src/store/commit.js

```diff
@@ -0,0 +1,33 @@
+import { indexGenerator } from "../api/fetch";
+
+export function fetchAll({ commit, rootState }, { collection, commitAction }) {
+  return new Promise((resolve, reject) => {
```

Comment:
Putting the boundary of async function to promise here doesn't make much sense to me. You can just treat `fetchAll` as a Promise in the different store files if you make it an async function.